chore: Remove old blocks from p2p

PhilWindle opened the pull request:
This PR updates the P2P subsystem to use the new `L2BlockNew` struct. The `L2Block` struct is received over the block stream and then immediately converted, meaning it will be simple to later change the block stream to work with the new block type.

spalladino reviewed (approved):
That was surprisingly straightforward!

PhilWindle committed:
chore: Remove old blocks from p2p

This PR updates the P2P subsystem to use the new `L2BlockNew` struct. The `L2Block` struct is received over the block stream and then immediately converted, meaning it will be simple to later change the block stream to work with the new block type.
